core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -50,14 +52,20 @@ class OffsetCheckpoint(val file: File) extends Logging {
           writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
           writer.newLine()
         }
-      
+
         // flush the buffer and then fsync the underlying file
         writer.flush()
         fileOutputStream.getFD().sync()
+      } catch {
+        case e: FileNotFoundException =>
+          if (FileSystems.getDefault.isReadOnly) {
+            fatal("Halting writes to offset checkpoint file because the underlying file system is inaccessible : ", e)
+            Runtime.getRuntime.halt(1)
+          }
```

Comment:
We should rethrow the exception in case the `if` statement evaluates to false, right?

Another question I have is whether this is the right place to halt the runtime. In an ideal world, we would throw an appropriate exception from classes such as this and halt the runtime in a centralised place somewhere (so that we could override the behaviour during tests). I spent a good amount of time tracking down a problem in Jenkins where the test suit would just die halfway due to a halt call in ReplicaManager which, in turn, was due to an exception being thrown from this class (https://issues.apache.org/jira/browse/KAFKA-3063). It's not an ideal world, however, so perhaps this is the best we can do for now.